src/System.Management.Automation/engine/Modules/PSModuleInfo.cs

```diff
@@ -400,13 +400,18 @@ public IEnumerable<string> Tags
             get { return _tags; }
         }
 
-        private readonly List<string> _tags = new List<string>();
+        private List<string> _tags = new List<string>();
 
         internal void AddToTags(string tag)
         {
             _tags.Add(tag);
```

Comment:
Seems like an alternate fix is for this line to check that `_tags` doesn't already contain the tag to add as duplicates don't make sense anyways.

Reply:
I believe my initial idea was `private readonly HashSet<string> _tags = new HashSet<string>(StringComparer.OrdinalIgnoreCase);` but it wouldn't break the reference.

```powershell
New-Item -ItemType File "tagsDemo.psm1"
New-ModuleManifest -Path "tagsDemo.psd1" -RootModule "tagsDemo.psm1" -FunctionsToExport '*' -Tags "hello"

$t = Test-ModuleManifest ./tagsDemo.psd1
New-ModuleManifest -Path "tagsDemo.psd1" -RootModule "tagsDemo.psm1" -FunctionsToExport '*' -Tags "world"

$t2 = Test-ModuleManifest ./tagsDemo.psd1

$t.Tags

# output
hello
world
```

Whether that's a real scenario or not, idk.

Reply:
You don't need a HashSet, `List` has a `Contains()` method you can just check against.  You also probably need to clear the `_tags` somewhere so they aren't accumulated across usage.

Reply:
> You also probably need to clear the `_tags` somewhere so they aren't accumulated across usage.

Isn't that the current state of the PR?

Breaking the `_tags` reference during shallow clone of cached module will fix both the original issue and the scenario above where the manifest has changed between tests.

> You don't need a HashSet, `List` has a `Contains()` method you can just check against.

`Contains()` wouldn't use case-insensitive comparison. Both alternatives are really solving a different issue though as it changes behavior by deduplicating tags intentionally set in a manifest-file.

I'll happily include it or provide a new PR for that as well if discussion, but imo. that would use case-insensitive comparison. 🙂

Reply:
> If I only clear the existing list it would solve the original issue, but not the sample above where the manifest-file has changed and you try to compare two manifest from different points in time.
> 
> Breaking the `_tags` reference during shallow clone of the cached entry will fix both the original issue and the scenario above.

Ignore this. Running `Test-ModuleManifest` against different versions of the same manifest-file is broken multiple places due to the cached copy, so fixing tags alone achieve much by itself. 🙂 

I can update the PR to only deduplicate.